public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -419,6 +421,7 @@ export class QueryEditorRow<TQuery extends DataQuery> extends PureComponent<Prop
               )}
               {editor}
             </ErrorBoundaryAlert>
+            {data && data.error && data.error.refId === query.refId && <QueryErrorAlert error={data.error} />}
```

Comment:
Can you extract `data && data.error && data.error.refId === query.refId` to a variable?